crates/web-plugins/didcomm-messaging/protocols/discover-features/src/handler.rs

```diff
@@ -0,0 +1,294 @@
+                                                    let id = format!("{id}.{minor}");
+                                                    disclosed_protocols.insert(id.to_string());
+                                                });
+                                        }
+                                    }
+                                }
+
+                                None => return Err(DiscoveryError::MalformedBody.into_response()),
+                            }
+                        } else {
+                            return Err(DiscoveryError::FeatureNOTSupported.into_response());
+                        }
+                    }
+                    None => return Err(DiscoveryError::MalformedBody.into_response()),
+                }
+            }
+
+            // build response body
+            let msg = build_response(disclosed_protocols);
+            Ok(Some(msg))
+        } else {
+            return Err(DiscoveryError::QueryNotFound.into_response());
+        }
+    } else {
+        let msg = build_response(disclosed_protocols);
+        Ok(Some(msg))
+    }
+}
+fn build_response(disclosed_protocols: HashSet<String>) -> Message {
+    let mut body = Disclosures::new();
+    for id in disclosed_protocols.iter() {
+        let content = DisclosuresContent {
+            feature_type: "protocol".to_string(),
+            id: id.to_owned(),
+            roles: None,
+        };
+        let content = json!(content);
+
+        body.disclosures.push(content);
+    }
+
+    let id = Uuid::new_v4().urn().to_string();
+    let msg = Message::build(id, DISCOVER_FEATURE.to_string(), json!(body)).finalize();
+
+    msg
+}
+#[cfg(test)]
+mod test {
+
+    use std::{sync::Arc, vec};
+
+    use did_utils::didcore::Document;
+    use didcomm::Message;
+    use keystore::tests::MockKeyStore;
+    use serde_json::json;
+    use shared::{
+        constants::QUERY_FEATURE,
+        repository::tests::{MockConnectionRepository, MockMessagesRepository},
+        state::{AppState, AppStateRepository},
+    };
+    use uuid::Uuid;
+
+    use crate::model::Queries;
+
+    use super::handle_query_request;
+    const MEDIATION: &str = "https://didcomm.org/coordinate-mediation/2.0";
+
+    pub fn setup() -> Arc<AppState> {
+        let public_domain = String::from("http://alice-mediator.com");
+
+        let diddoc = Document::default();
+
+        let repository = AppStateRepository {
+            connection_repository: Arc::new(MockConnectionRepository::from(vec![])),
+            message_repository: Arc::new(MockMessagesRepository::from(vec![])),
+            keystore: Arc::new(MockKeyStore::new(vec![])),
+        };
+
+        let state = Arc::new(AppState::from(
+            public_domain,
+            diddoc,
+            Some(vec![
+                "https://didcomm.org/coordinate-mediation/2.0/mediate-request".to_string(),
+            ]),
+            Some(repository),
```

Comment:
You can simply enter the `None` parameter if you are not using the App repository.